File: eSalon/eSalon.Model/Recenzija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eSalon.Model
{
    public class Recenzija
    {
        public int RecenzijaId { get; set; }

        public int KorisnikId { get; set; }

        public int UslugaId { get; set; }

        public string Komentar { get; set; } = null!;

        public DateTime DatumDodavanja { get; set; }

        public int BrojLajkova { get; set; }

        public int BrojDislajkova { get; set; }

        public string? KorisnickoIme { get; set; }
        public string? NazivUsluge { get; set; }
        public int BrojOdgovora { get; set; }
        public bool JeLajkOdKorisnika { get; set; } = false;
        public bool JeDislajkOdKorisnika { get; set; } = false;

    }
}

File: eSalon/UI/esalon_mobile/lib/models/recenzija.dart
import 'package:json_annotation/json_annotation.dart';

part 'recenzija.g.dart';

@JsonSerializable()
class Recenzija {
  int? recenzijaId;
  int korisnikId;
  int uslugaId;
  String komentar;
  DateTime datumDodavanja;
  int brojLajkova;
  int brojDislajkova;
  String? korisnickoIme;
  String? nazivUsluge;
  int brojOdgovora;
  bool jeLajkOdKorisnika;
  bool jeDislajkOdKorisnika;

  Recenzija(
    this.recenzijaId,
    this.korisnikId,
    this.uslugaId,
    this.komentar,
    this.datumDodavanja,
    this.brojLajkova,
    this.brojDislajkova,
    this.korisnickoIme,
    this.nazivUsluge,
    this.brojOdgovora,
    this.jeLajkOdKorisnika, 
    this.jeDislajkOdKorisnika,  
  );
  
  factory Recenzija.fromJson(Map<String, dynamic> json) => _$RecenzijaFromJson(json);

  Map<String, dynamic> toJson() => _$RecenzijaToJson(this);
}